### ISSUE TITLE
roxygen2 doesn't respect collate order

### ISSUE THREAD

**Issue by wch** [klutometis/roxygen#360]
Suppose you have two files, `a.R` and `b.R`:

```R
## a.R ##
#' @include b.R
NULL

#' Function a
#' @rdname funb
funa <- function() NULL
```

```R
## b.R ##
#' Function b
funb <- function() NULL
```

Then the resulting .Rd file should have `funb` in the header, and `funb` should be listed first. But instead, `funa` is in the header and is listed first.

Here are the contents of `funb.Rd`:

```R
% Generated by roxygen2 (4.1.1): do not edit by hand
% Please edit documentation in R/a.R, R/b.R
\name{funa}
\alias{funa}
\alias{funb}
\title{Function a}
\usage{
funa()

funb()
}
\description{
Function a

Function b
}
```

**Comment by cbeleites** [comment]
I'm struggling with this, too. 
```r
## b.R ##
@describeIn funa
``` 
solves some of the trouble, but e.g. if both functions give `@details`, the order within the details section is still wrong.

**Comment by Geoff99** [comment]
Hi @wch,
I missed this when you first posted it, but I rather *suspect* (aka I haven't actually tested your example explicitly, though I have with my own similar min working example) that it has the same root cause as issue #323, namely that: 

  + although roxygen2 (and devtools) *source* code in collate order (essential obviously), 
  + as at when I last looked, roxygen2 actually **parsed** the R code files in alphabetical order.  

If so, it should be resolved by pull request #324.
Regards
Geoff

**Comment by hadley** [comment]
Fixed in
